In [1]:
import pandas as pd

In [5]:
pd.read_csv("data/moradores.csv", sep = ";")

/tmp/ipykernel_8135/68921856.py:1: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  pd.read_csv("data/moradores.csv", sep = ";")


,A01nficha,A01uf,localidade,setor_distrito,morador_id,index,idade_calculada,E02_1,E02_2,E02_3,...,I22_2_r,I23,renda_ind,renda_ind_r,infla_ipcadf,id_genero,orientacao_sx,lgbtqiam,escolaridade,peso_mor
0,1,53,5330,35,00001-1,1,49,99999,99999,49,...,0,50,0,0,"0,989671705",1,2,2,5,"78,62021308"
1,2,53,5320,24,00002-1,1,35,1988-10-17,99999,99999,...,99999,6,99999,99999,"0,982195674",1,2,2,7,"148,8188948"
2,2,53,5320,24,00002-2,2,41,1982-08-28,99999,99999,...,99999,6,99999,99999,"0,982195674",1,2,2,7,"212,8951827"
3,2,53,5320,24,00002-3,3,5,2019-01-04,99999,99999,...,99999,99999,0,0,"0,982195674",99999,99999,99999,1,"341,5004303"
4,2,53,5320,24,00002-4,4,3,2020-09-25,99999,99999,...,99999,99999,0,0,"0,982195674",99999,99999,99999,1,"244,9026476"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
69537,24844,53,5319,23,24844-2,2,11,99999,99999,11,...,0,10,0,0,"0,974884044",99999,99999,99999,2,"14,5461749"
69538,24844,53,5319,23,24844-3,3,7,99999,99999,7,...,99999,99999,0,0,"0,974884044",99999,99999,99999,2,"13,59904416"
69539,24844,53,5319,23,24844-4,4,7,99999,99999,7,...,99999,99999,0,0,"0,974884044",99999,99999,99999,2,"15,46400694"
69540,24845,52,5249,46,24845-1,1,97,99999,99999,97,...,0,15,1420,"1429,957381","0,993036589",1,2,2,8,"10,61778243"


Parte 2 — Exercícios de Programação

Exercício 1.1 — Lendo os dados

In [7]:
import pandas as pd

moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")
print(moradores.head())

   A01nficha  A01uf  localidade  setor_distrito morador_id  index  \
0          1     53        5330              35    00001-1      1   
1          2     53        5320              24    00002-1      1   
2          2     53        5320              24    00002-2      2   
3          2     53        5320              24    00002-3      3   
4          2     53        5320              24    00002-4      4   

   idade_calculada       E02_1  E02_2  E02_3  ...  I22_2_r    I23 renda_ind  \
0               49       99999  99999     49  ...      0.0     50         0   
1               35  1988-10-17  99999  99999  ...  99999.0      6     99999   
2               41  1982-08-28  99999  99999  ...  99999.0      6     99999   
3                5  2019-01-04  99999  99999  ...  99999.0  99999         0   
4                3  2020-09-25  99999  99999  ...  99999.0  99999         0   

   renda_ind_r infla_ipcadf  id_genero  orientacao_sx lgbtqiam  escolaridade  \
0          0.0     0.989672   

/tmp/ipykernel_8135/921568067.py:3: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")


In [8]:
moradores.shape

(69542, 134)

há 69.542 linhas e 134 colunas no arquivo

Exercício 1.2 — Selecionando colunas

In [9]:
import pandas as pd

moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")

colunas = ["morador_id", "localidade", "idade_calculada", "id_genero", "escolaridade", "renda_ind"]
print(moradores[colunas])

      morador_id  localidade  idade_calculada  id_genero  escolaridade  \
0        00001-1        5330               49          1             5   
1        00002-1        5320               35          1             7   
2        00002-2        5320               41          1             7   
3        00002-3        5320                5      99999             1   
4        00002-4        5320                3      99999             1   
...          ...         ...              ...        ...           ...   
69537    24844-2        5319               11      99999             2   
69538    24844-3        5319                7      99999             2   
69539    24844-4        5319                7      99999             2   
69540    24845-1        5249               97          1             8   
69541    24845-2        5249               51          1             8   

       renda_ind  
0              0  
1          99999  
2          99999  
3              0  
4              0

/tmp/ipykernel_8135/3392364638.py:3: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")


In [10]:
import pandas as pd

moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")

colunas = ["morador_id", "localidade", "idade_calculada", "id_genero", "escolaridade", "renda_ind", "peso_mor"]
print(moradores[colunas])

      morador_id  localidade  idade_calculada  id_genero  escolaridade  \
0        00001-1        5330               49          1             5   
1        00002-1        5320               35          1             7   
2        00002-2        5320               41          1             7   
3        00002-3        5320                5      99999             1   
4        00002-4        5320                3      99999             1   
...          ...         ...              ...        ...           ...   
69537    24844-2        5319               11      99999             2   
69538    24844-3        5319                7      99999             2   
69539    24844-4        5319                7      99999             2   
69540    24845-1        5249               97          1             8   
69541    24845-2        5249               51          1             8   

       renda_ind    peso_mor  
0              0   78.620213  
1          99999  148.818895  
2          99999  

/tmp/ipykernel_8135/1512070687.py:3: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")


Exercício 1.3 — Contando moradores por domicílio

In [12]:
import pandas as pd

domicilios = pd.read_excel("data/domicilios.xlsx")

for _, linha in domicilios.iterrows():
    print(f"Domicílio {linha['A01nficha']}: {linha['A01npessoas']} moradores, {linha['A01ncriancas']} crianças")

A saída de streaming foi truncada nas últimas 5000 linhas.
Domicílio 19846: 4 moradores, 1 crianças
Domicílio 19847: 3 moradores, 1 crianças
Domicílio 19848: 2 moradores, 0 crianças
Domicílio 19849: 2 moradores, 0 crianças
Domicílio 19850: 3 moradores, 0 crianças
Domicílio 19851: 2 moradores, 0 crianças
Domicílio 19852: 3 moradores, 0 crianças
Domicílio 19853: 4 moradores, 1 crianças
Domicílio 19854: 2 moradores, 0 crianças
Domicílio 19855: 2 moradores, 0 crianças
Domicílio 19856: 5 moradores, 1 crianças
Domicílio 19857: 2 moradores, 0 crianças
Domicílio 19858: 4 moradores, 2 crianças
Domicílio 19859: 1 moradores, 0 crianças
Domicílio 19860: 5 moradores, 0 crianças
Domicílio 19861: 1 moradores, 0 crianças
Domicílio 19862: 3 moradores, 1 crianças
Domicílio 19863: 1 moradores, 0 crianças
Domicílio 19864: 8 moradores, 4 crianças
Domicílio 19865: 5 moradores, 1 crianças
Domicílio 19866: 5 moradores, 0 crianças
Domicílio 19867: 1 moradores, 0 crianças
Domicílio 19868: 2 moradores, 0 criança

In [19]:
import pandas as pd

domicilios = pd.read_excel("data/domicilios.xlsx")

maior_quantidade_de_moradores = 0
maior_quantidade_de_criancas = 0
domicilio_com_mais_moradores = 0
domicilio_com_mais_criancas = 0

for _, linha in domicilios.iterrows():

    if (int(linha['A01npessoas']) > maior_quantidade_de_moradores):
        maior_quantidade_de_moradores = int(linha['A01npessoas'])
        domicilio_com_mais_moradores = int(linha["A01nficha"])
        # print(f"Domicílio {linha['A01nficha']}: {linha['A01npessoas']} moradores, {linha['A01ncriancas']} crianças")

    if (int(linha['A01ncriancas']) > maior_quantidade_de_criancas):
        maior_quantidade_de_criancas = int(linha['A01ncriancas'])
        domicilio_com_mais_criancas = int(linha["A01nficha"])

print(f"domicílio com mais moradores: {domicilio_com_mais_moradores}, com {maior_quantidade_de_moradores} moradores")
print(f"domicílio com mais crianças: {domicilio_com_mais_criancas}, com {maior_quantidade_de_criancas} crianças")


domicílio com mais moradores: 5108, com 14 moradores
domicílio com mais crianças: 5108, com 11 crianças


Exercício 1.4 — Exibindo idades

In [20]:
import pandas as pd

moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")

adultos = moradores[moradores["idade_calculada"] != 99999]
Os arquivos não tratam de idadeprint(adultos[["morador_id", "idade_calculada"]])

/tmp/ipykernel_8135/4240246334.py:3: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")


      morador_id  idade_calculada
0        00001-1               49
1        00002-1               35
2        00002-2               41
3        00002-3                5
4        00002-4                3
...          ...              ...
69537    24844-2               11
69538    24844-3                7
69539    24844-4                7
69540    24845-1               97
69541    24845-2               51

[69542 rows x 2 columns]


Os arquivos não tratam de idade declarada, apenas de idade calculada.

Como a quantidade de linhas do filtro é igual à quantidade total de linhas (69542), conclui-se que nenhum morador possui idade calculada de 99999.

Alguém porventura poderia ter idade calculada de 99999 caso a data de nascimento dessa pessoa fosse desconhecida, assim sua idade não poderia ser calculada e receberia o valor 99999, valor este que é um valor especial que indica que o valor não é válido naquele caso.

Exercício 2.1 — Calculando a idade média


In [22]:
import pandas as pd

moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")

idades = moradores[moradores["idade_calculada"] != 99999]["idade_calculada"].tolist()

soma = 0
for idade in idades:
    soma = soma + idade

media = soma / len(idades)
print(f"Total de moradores com idade declarada: {len(idades)}")
print(f"Soma das idades: {soma}")
print(f"Média de idade: {media:.1f} anos")

/tmp/ipykernel_8135/3564209607.py:3: DtypeWarning: Columns (12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  moradores = pd.read_csv("data/moradores.csv", sep=";", decimal=",", encoding="utf-8-sig")


Total de moradores com idade declarada: 69542
Soma das idades: 2518823
Média de idade: 36.2 anos
